### ISSUE TITLE
Randomness in Arena

### ISSUE THREAD

**Issue by bhansconnect** [evg-tyurin/alpha-nagibator#4]
Quick questions about Arena comparison of networks.

If the temperature is set to 0 and there is no Dirichlet noise added in during Arena games, what stops the neural networks from playing the same game over and over again? Where does the randomness come in that makes every game unique?

Is the game only random when the MCTS has 2 moves with the same count? I feel like that would be pretty rare. I am just trying to understand how you compare networks and make things random/unique. Any comments would be greatly appreciated.

**Comment by evg-tyurin** [comment]
Yes, randomness occurs in two different places in the code. When I run competitions in Arena I observe zero to 15-20 identical games or even more depending on unknown conditions. I think that more identical games occur if network is overfitted.

**Comment by bhansconnect** [comment]
Can you point out those 2 locations for me?

**Comment by evg-tyurin** [comment]
https://github.com/evg-tyurin/alpha-nagibator/blob/48b2ebd3ca272f388c13277297edbb60d98eb64b/MCTS.py#L75

https://github.com/evg-tyurin/alpha-nagibator/blob/48b2ebd3ca272f388c13277297edbb60d98eb64b/MCTS.py#L225
